cpp/arcticdb/storage/azure/azure_storage.cpp

```diff
@@ -47,6 +47,61 @@ namespace fg = folly::gen;
 
 namespace detail {
 
+// TODO: fix this temporary workaround to read error code. azure-sdk-cpp client sometimes doesn't properly set the error code.
+//  This issue has been raised on the sdk repo. Once fixed, we should no longer need the following function and would just read e.ErrorCode.
+std::string get_error_code(const Azure::Core::RequestFailedException& e) {
+    auto error_code = e.ErrorCode;
+
+    if(error_code.empty() && e.RawResponse ) {
+        auto headers_map = e.RawResponse->GetHeaders();
+        if(headers_map.find("x-ms-error-code") != headers_map.end()) {
+            error_code = headers_map["x-ms-error-code"];
```

Comment:
You've just found the key, so why look it up again?

```
if (auto ec = headers_map.find("x-ms-error-code") ; ec != headers_map.end()) {
  error_code = *ec;
}
```

Reply:
Fixed in [4ea3442](https://github.com/man-group/ArcticDB/pull/1344/commits/4ea3442882bfdc4001ff117531804a8ba28a8d08)